src/clusterfuzz/_internal/bot/fuzzers/syzkaller/runner.py

```diff
@@ -70,14 +73,37 @@ def get_config():
 
 def get_cover_file_path():
   """Return location of coverage file for Syzkaller."""
-  return os.path.join(get_work_dir(), 'coverfile')
+  file_path = os.path.join(get_work_dir(), 'coverfile')
+  if not os.path.isfile(file_path):
+    open(file_path, 'w+').close()
+
+  logs.info(f'Rawcover file will be stored in {file_path}')
+  return file_path
 
 
 def get_runner(fuzzer_path):
   """Return a syzkaller runner object."""
   return AndroidSyzkallerRunner(fuzzer_path)
 
 
+def get_syzkaller_pid() -> int:
+  """Get pid of syzkaller.
+
+  Assumes syzkaller is the only child process of clusterfuzz.
+
+  Returns:
+    syzkaller pid. 0 if not found.
+  """
+  children_procs = psutil.Process(os.getpid()).children()
```

Comment:
This is a fragile assumption and could easily break in the future. 

We can pipe the pid in a much better way, if we make a simple refactoring  to new_process:

- make `_wait_process` public (i.e. `new_process.wait_process`)
- Expose `pid` in `new_process.ChildProcess`
- Call `ProcessRunner.run` instead of `ProcessRunner.run_and_wait`, and grab the `pid`, start the thread to poll for coverage and then call `new_process.wait_process` on it similar to here: https://github.com/google/clusterfuzz/blob/5b8f9b9efb25e4291024eeacfc254162a34dc320/src/clusterfuzz/_internal/system/new_process.py#L347

Reply:
the reason I was not using this is because `process.communicate()` is a blocking call from my understanding. pid will not return until `syzkaller` is finished. We need a process to get `rawcover` while `syzkaller` process is running.

Reply:
Yes, please see my refactoring suggestion to resolve this.

Reply:
yeah this is something I considered, but didn't want to make too big of a refactor. I will take your suggestion and try to minimize the refactor size.